frontend/src/components/associations/settings/AssociationNamespaceSettings.tsx

```diff
@@ -0,0 +1,232 @@
+            const error = errorAsUnknown as AxiosError;
+            newToast({
+                message: `Erreur. Merci de réessayer ou de contacter les administrateurs si cela persiste. ${error.message}`,
+                level: ToastLevel.Error,
+            });
+        },
+    });
+
+    return (
+        <>
+            <Card>
+                <Card.Header>
+                    <Card.Title>Tags</Card.Title>
+                </Card.Header>
+            </Card>
+
+            {selectedNamespace && (
+                <NamespaceTagsModal
+                    namespace={selectedNamespace}
+                    closeModal={() => setSelectedNamespace(null)}
+                />
+            )}
+
+            <Pagination
+                apiKey={[
+                    "association.namespaces.list",
+                    { association: association.id },
+                ]}
+                apiMethod={api.namespaces.list}
+                render={(namespaces, paginationControl) => (
+                    <>
+                        <Row>
+                            {namespaces.map((namespace) => (
+                                <ListNamespaceTags
+                                    key={namespace.id}
+                                    namespace={namespace}
+                                    setSelectedNamespace={setSelectedNamespace}
+                                />
+                            ))}
+                        </Row>
+                        {paginationControl}
+                    </>
+                )}
+            />
+
+            <Formik
+                onSubmit={(values) => {
+                    createNamespace({
+                        name: values.namespaceName,
+                        scoped_to_model: "association",
+                        scoped_to_pk: association.id,
+                    });
+                }}
+                initialValues={{ namespaceName: "namespace" }}
+                validationSchema={Yup.object({
+                    namespaceName: Yup.string()
+                        .required("Ce champ est requis.")
+                        .min(
+                            2,
+                            "Un namespace doit avoir au minimum 2 caractères"
+                        ),
+                })}
+            >
+                {({ values, handleChange, errors, touched, handleSubmit }) => (
+                    <form onSubmit={handleSubmit}>
+                        <InputGroup>
+                            <Form.Control
+                                placeholder={"Nouveau namespace"}
+                                name={"namespaceName"}
+                                value={values.namespaceName}
+                                onChange={handleChange}
+                                isInvalid={
+                                    (errors.namespaceName &&
+                                        touched.namespaceName) ||
+                                    undefined
```

Comment:
why is there `undefined` here?
Doesn't it always evaluate to `false`? What's the point?

Reply:
`undefined || undefined == undefined`, I changed it to `(errors.namespaceName && touched.namespaceName) || false`